### PR TITLE
Fixed product back redirect navigate from customer view cart product

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

Fixed product back redirect navigate from customer view cart product
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: N/A

### Manual testing scenarios (*)

1. create new customer
2. create new product
3. add product into cart 
4. Admin -> Customers -> All Customers -> edit customer -> Shopping cart Tab -> click on product row -> from edit product click for back

Expected result (*)
back redirect  edit customer 

Actual result (*)
back redirect catalog/product/index

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ravi-chandra3197** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197, here is your [Magento instance](https://i-24089-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24089-2-3-develop.instances.magento-community.engineering/admin](https://i-24089-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ravi-chandra3197** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by ravi-chandra3197** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197, here is your [Magento instance](https://i-24089-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24089-2-3-develop.instances.magento-community.engineering/admin](https://i-24089-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/Back.php [hunk 1]
```diff
@@ -17,9 +17,21 @@ public function getButtonData()
...
             'class' => 'back',
             'sort_order' => 10
         ];
     }
+    /**
+     * Get URL for back
+     *
+     * @return string
+     */
+    public function getBackUrl()
```
Comment (VladimirZaets) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Button/Back.php]: Hi, @ravi-chandra3197, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't create new one public methods. Please make this method private.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5637](https://jira.corp.magento.com/browse/ENGCOM-5637) has been created to process this Pull Request
:eight_spoked_asterisk: @VladimirZaets, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @ravi-chandra3197 .

During testing we faced the issue

_**Problem:**_ Error  "Invalid security or form key. Please refresh the page." message occurs

**Steps to reproduce:**

1. Create new customer in storefront;
2. Create new product;
3. Add product into cart;
4. Go to Admin -> Customers -> All Customers -> edit customer -> Shopping cart Tab -> click on product ; 
5. From edit product click on "Back" button;

**Actual result:**
after

@ravi-chandra3197  Could you take a look?

Thanks!

**Comment by ravi-chandra3197** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197, here is your [new Magento instance](https://pr-24089.instances.magento-community.engineering).
Admin access: [https://pr-24089.instances.magento-community.engineering/admin](https://pr-24089.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by krnshah** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @krnshah. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @krnshah, here is your [new Magento instance](https://pr-24089.instances.magento-community.engineering).
Admin access: [https://pr-24089.instances.magento-community.engineering/admin](https://pr-24089.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by ravi-chandra3197** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197, here is your [new Magento instance](https://pr-24089.instances.magento-community.engineering).
Admin access: [https://pr-24089.instances.magento-community.engineering/admin](https://pr-24089.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by ravi-chandra3197** [comment]
Hello @engcom-Alfa 
I have update PR and resolve above issue can you check it. 

> Hi @ravi-chandra3197 .
> 
> During testing we faced the issue
> 
> _**Problem:**_ Error "Invalid security or form key. Please refresh the page." message occurs
> 
> **Steps to reproduce:**
> 
> 1. Create new customer in storefront;
> 2. Create new product;
> 3. Add product into cart;
> 4. Go to Admin -> Customers -> All Customers -> edit customer -> Shopping cart Tab -> click on product ;
> 5. From edit product click on "Back" button;
> 
> **Actual result:**
> after
> 
> @ravi-chandra3197 Could you take a look?
> 
> Thanks!

Hello @engcom-Alfa 
I have updated PR can you check it.

**Comment by VladimirZaets** [comment]
Hi @ravi-chandra3197. Please fix static tests. Thanks

**Comment by ravi-chandra3197** [comment]
> fix static tests

Hello @VladimirZaets 
I have fixed static tests can review it.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-5637](https://jira.corp.magento.com/browse/ENGCOM-5637) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**